src/main/java/hudson/plugins/build_timeout/impl/DeadlineTimeOutStrategy.java

```diff
@@ -87,7 +87,7 @@ public long getTimeOut(AbstractBuild<?, ?> build, BuildListener listener) throws
         }
 
         listener.getLogger().println(
-                Messages.DeadlineTimeOutStrategy_NextDeadline(TIMESTAMP_FORMAT.format(deadlineTimestamp
+                Messages.DeadlineTimeOutStrategy_NextDeadline(new SimpleDateFormat(TIMESTAMP_FORMAT_PATTERN).format(deadlineTimestamp
```

Comment:
DeadlineTimeOutStrategy.java:89, STCAL_INVOKE_ON_STATIC_DATE_FORMAT_INSTANCE, Priority: Normal

Call to method of static java.text.DateFormat in hudson.plugins.build_timeout.impl.DeadlineTimeOutStrategy.getTimeOut(AbstractBuild, BuildListener)

As the JavaDoc states, DateFormats are inherently unsafe for multithreaded use. The detector has found a call to an instance of DateFormat that has been obtained via a static field. This looks suspicous.

For more information on this see Sun Bug #6231579 and Sun Bug #6178997.